torchvision/utils.py

```diff
@@ -210,3 +210,59 @@ def draw_bounding_boxes(
             draw.text((bbox[0], bbox[1]), labels[i], fill=color, font=txt_font)
 
     return torch.from_numpy(np.array(img_to_draw)).permute(2, 0, 1)
+
+
+@torch.no_grad()
+def draw_segmentation_masks(
+    image: torch.Tensor,
+    masks: torch.Tensor,
+    colors: Optional[List[Union[str, Tuple[int, int, int]]]] = None,
+) -> torch.Tensor:
+
+    """
+    Draws segmentation masks on given image.
+    The values of the input image should be uint8 between 0 and 255.
+
+    Args:
+        image (Tensor): Tensor of shape (C x H x W)
+        masks (Tensor): Tensor of shape (H, W). Each containing predicted class.
```

Comment:
I thought it would seem odd to pass probabilities of each. It would definitely give `num_classes`.

I had a small concern when user's outputs aren't probabilities (Though in most semantic segmentation cases we will have such output).

Quite unsure here. I'm not sure what would be consistent with `draw_bounding_boxes`.

`draw_bounding_boxes` with FRCNN

```
>>> frcnn = torchvision.models.detection.fasterrcnn_resnet50_fpn(pretrained=False, pretrained_backbone=False)
>>> frcnn = frcnn.eval()
>>> image = torch.randn(3, 224, 224)
>>> out = frcnn([image])
>>> boxes = out[0]["boxes"]
>>> image = torchvision.transforms.ConvertImageDtype(torch.uint8)(image)
>>> result = draw_bounding_boxes(image=image, boxes=boxes, fill=(50, 100, 100, 127), colors=["black", "black", "black"])
```

I think `draw_segmentation_masks` will work like

```
fcn = fcn_resnet50(pretrained=False, num_classes=21)
fcn = fcn.eval()
image = torch.ones(3, 224, 224)
fcn_out = fcn([image])['out'][0]
result = draw_segmentation_masks(image, masks, colors=colors)
```

This looks fine and I think is consistent :smiley:

Let me refactor to this. This will slow the `num_classes` issue and assumption too.

Reply:
Great! Looks good and consistent to me. :)